src/Core/ViewHelper/AbstractConditionViewHelper.php

```diff
@@ -318,8 +286,58 @@ final public function convert(TemplateCompiler $templateCompiler): array
                 }
             }
         }
-        if (!$thenChildEncountered && $elseIfCounter === 0 && !$elseChildEncountered && !isset($node->getArguments()['then'])) {
-            if (!isset($node->getArguments()['else']) && $node->getChildNodes() === []) {
+
+        $accumulatedArgumentInitializationCode = '';
+        $arguments = $node->getArguments();
+        $argumentDefinitions = $node->getArgumentDefinitions();
+        foreach ($argumentDefinitions as $argumentName => $argumentDefinition) {
+            if (!array_key_exists($argumentName, $arguments)) {
+                // Argument *not* given to VH, use default value
+                $defaultValue = $argumentDefinition->getDefaultValue();
+                $argumentInitializationCode .= sprintf(
+                    '\'%s\' => %s,' . chr(10),
+                    $argumentName,
+                    is_array($defaultValue) && empty($defaultValue) ? '[]' : var_export($defaultValue, true),
+                );
+            } elseif ($arguments[$argumentName] instanceof NodeInterface) {
+                // Argument *is* given to VH and is a node, resolve
+                $converted = $arguments[$argumentName]->convert($templateCompiler);
+                $accumulatedArgumentInitializationCode .= $converted['initialization'];
+
+                if ($argumentName === 'then' || $argumentName === 'else') {
+                    $argumentInitializationCode .= sprintf(
+                        '\'__%s\' => %s,' . chr(10),
+                        $argumentName,
+                        'function () use ($renderingContext) { return ' . $converted['execution'] . ';}',
```

Comment:
thill will break access to the variable of the parent scope. We could try with arrow function. I'll try to create a testcase to reproduce the error and add my implementation with arrow function